csharp/ql/lib/semmle/code/csharp/exprs/Call.qll

```diff
@@ -290,7 +303,54 @@ private predicate isValidExplicitParamsType(Parameter p, Type t) {
  * ```
  */
 class MethodCall extends Call, QualifiableExpr, LateBindableExpr, @method_invocation_expr {
-  override Method getTarget() { expr_call(this, result) }
+  override Method getTarget() {
+    expr_call(this, result)
+    or
+    result = this.getACandidateTarget()
+  }
+
+  private Method getACandidateTarget() {
+    not exists(Method m | expr_call(this, m)) and
+    exists(string name, ValueOrRefType alternativeQualifierType |
+      invocation_member_name(this, name) and
+      alternativeQualifierType =
+        this.getQualifier()
+            .getType()
+            .(ValueOrRefType)
+            .getABaseType*()
```

Comment:
Including base types here somehow seems related to dispatching. Could this be a potential performance killer? Would it make sense to initially only look at type alternatives without taking base types into account?

Reply:
The test cases show that including the base classes is needed. At the same time, it can easily be a performance killer.